invenio_vocabularies/services/components.py

```diff
@@ -49,3 +49,12 @@ def create(self, identity, data=None, record=None, **kwargs):
         # We create the PID after all the data has been initialized. so that
         # we can rely on having the 'id' and type set.
         self.service.record_cls.pid.create(record)
+
+
+class ModelPIDComponent(PIDComponent):
+    """Model PID registration component."""
+
+    def update(self, identity, data=None, record=None, **kwargs):
+        """Update an existing record."""
+        record.pop("pid", None)
+        record.pop("id", None)
```

Comment:
Question: where does the `id` field come from in the record metadata?

Reply:
Because it is an update and the `result_item.data` does a `marshmallow.dump`, which has to contain the id.